src/main/java/com/epam/aidial/core/storage/credential/CredentialProviderFactory.java

```diff
@@ -0,0 +1,17 @@
+package com.epam.aidial.core.storage.credential;
+
+import com.epam.aidial.core.storage.StorageProvider;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class CredentialProviderFactory {
+    public static CredentialProvider create(String providerName, String identity, String credential) {
+        StorageProvider provider = StorageProvider.from(providerName);
+        return switch (provider) {
+            case S3, AZURE_BLOB -> new DefaultCredentialProvider(identity, credential);
+            case GOOGLE_CLOUD_STORAGE -> new GcpCredentialProvider(credential);
```

Comment:
why you remove the "old" one with static credentials for gcp?

Reply:
It's not removed but replace with new one.